onnxruntime/test/providers/provider_test_utils.h

```diff
@@ -529,7 +530,27 @@ class OpTester {
       OrtValue value;
       value.Init(p_tensor.release(), DataTypeImpl::GetType<Tensor>(),
                  DataTypeImpl::GetType<Tensor>()->GetDeleteFunc());
-      data.push_back(Data(NodeArg(name, &type_proto), std::move(value), optional<float>(), optional<float>(), sort_output));
+      auto node_arg = NodeArg(name, &type_proto);
+      if (dim_params && !(dim_params->empty())) {
+        // If dim_params presents, configure node_arg's dim value based on dim_params, which supports symbolic dim and dim broadcast.
+        auto& dim_params_data = *dim_params;
+        onnx::TensorShapeProto new_shape;
+
+        // currently hard-code the reserved symbolic names.
+        // TODO: when the list grows longer, consider move it to a better place.
+        const static std::unordered_set<std::string> reserved_symbolic{"batch", "seq"};
```

Comment:
shouldn't need to do this..

Reply:
I prefer {"4", "2", "1", "seq"} mostly because when in a case to figure out the designed tensor shape, I think {"4", "2", "1", "seq"} is more straight forward than {4, 2, 1, 3} + {"", "", "", "seq"}. The choice is mostly for readability. And since it is a test, we can afford to provide a bit more redundant data :-)

---
In reply to: [418774393](https://github.com/microsoft/onnxruntime/pull/3743#discussion_r418774393) [](ancestors = 418774393)